context.go

```diff
@@ -417,11 +419,22 @@ func (ctx *Context) GetSecureCookie(key string) (string, bool) {
 	return ctx.GetSuperSecureCookie(defaultCookieSecret, key)
 }
 
+// generateNonce generates random bytes from a secure source.
+func (ctx *Context) generateNonce() error {
+	_, err := rand.Read(ctx.nonce)
+
+	return err
+}
+
 // SetSuperSecureCookie sets given cookie value to response header with secret string.
 func (ctx *Context) SetSuperSecureCookie(secret, name, value string, others ...interface{}) {
-	m := md5.Sum([]byte(secret))
-	secret = hex.EncodeToString(m[:])
-	text, err := com.AESEncrypt([]byte(secret), []byte(value))
+
+	// Ideally we'd want to use PBKDF2 here.
+	// Stick with SHA256 for now so we don't break the API.
+	// This method would need to take in a salt and pass it to PBKDF2 if we want to switch to PBKDF2.
+	hash := sha256.Sum256([]byte(secret))
+	secret = hex.EncodeToString(hash[:])
+	text, err := com.AESEncrypt([]byte(secret), ctx.nonce, []byte(value)) // The nonce should really be unique per cookie.
```

Comment:
It should be as you say, unless there are dependencies on other libraries/packages that are expecting it in this way. Then they'd all need to be changed. @Unknwon would know more about that than me.